File: k-closest-points-to-origin/k-closest-points-to-origin.py
from heapq import *
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        minHeap = []
        for point in points: 
            x, y = point[0], point[1]
            distance = sqrt((x ** 2) + (y ** 2))
            minHeap.append((distance, x, y))
        heapify(minHeap)

        result = []
        for _ in range(k):
            (distance, x, y) = heappop(minHeap)
            result.append([x, y])
        
        return result 

File: k-closest-points-to-origin/k-closest-points-to-origin.java
class Solution {
    public int[][] kClosest(int[][] points, int k) {
        int[][] result = new int[k][2];
        PriorityQueue<int[]> pq = new PriorityQueue<int[]>(
            (p1, p2) -> getEuclideanDistance(p1).compareTo(getEuclideanDistance(p2))
        );

        // Calculate distance from origin for each point + Add to Priority Queue
        // PQ orders points by ascending distance
        for (int[] point : points) {
            pq.add(point);
        }

        // Add K closest points to origin to result
        for (int i = 0; i < k; i++) {
            result[i] = pq.poll();
        }

        return result;
    }

    public Double getEuclideanDistance(int[] point) {
        return Math.sqrt(Math.pow(point[0], 2) + Math.pow(point[1], 2));
    }
}